thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -241,16 +235,25 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
         {
             if (builder)
             {
-                builder->finish(metadata, &crc, maxRecordSizeSeen);
-                if (!isTLK)
+                // Clear out builder before merging builder stats into inactive stats
+                // so that gatherActiveStatistics doesn't also merge builder stats.
+                Owned <IKeyBuilder> tmpBuilder;
+                {
+                    CriticalBlock b(builderCS);
+                    tmpBuilder.setown(builder.getClear());
+                }
+                if (tmpBuilder)
                 {
-                    duplicateKeyCount = builder->getDuplicateCount();
-                    numLeafNodes = builder->getNumLeafNodes();
-                    numBranchNodes = builder->getNumBranchNodes();
-                    numBlobNodes = builder->getNumBlobNodes();
-                    offsetBranches = builder->getOffsetBranches();
-                    branchMemorySize = builder->getBranchMemorySize();
-                    leafMemorySize = builder->getLeafMemorySize();
+                    tmpBuilder->finish(metadata, &crc, maxRecordSizeSeen);
+                    if (!isTLK)
+                    {
+                        duplicateKeyCount = tmpBuilder->getDuplicateCount();
+                        offsetBranches = tmpBuilder->getOffsetBranches();
+                        branchMemorySize = tmpBuilder->getBranchMemorySize();
+                        leafMemorySize = tmpBuilder->getLeafMemorySize();
```

Comment:
future: for next JIRA, but these should be added as stats to indexWriteActivityStatistics, and collected via builder in a common way, and manual serialization/deserialization here and in master should be removed.

Reply:
Future: https://hpccsystems.atlassian.net/browse/HPCC-33279